operator-common/src/main/java/io/strimzi/operator/common/operator/resource/StatusUtils.java

```diff
@@ -244,22 +244,25 @@ public static Condition getPausedCondition() {
      * Validate the Custom Resource. This should log at the WARN level (rather than throwing) if the resource can safely
      * be reconciled (e.g. it merely using deprecated API).
      *
-     * @param <T>               Custom Resource type
-     * @param <P>               Custom Resource spec type
-     * @param <S>               Custom Resource status type
+     * @param <T>                   Custom Resource type
+     * @param <P>                   Custom Resource spec type
+     * @param <S>                   Custom Resource status type
      *
-     * @param reconciliation    The reconciliation
-     * @param resource          The custom resource
+     * @param reconciliation        The reconciliation
+     * @param resource              The custom resource
+     * @param printResourceName     Flag indicating whether the warning messages should contain the resource
+     *                              identification or not. This is useful when the message is logged as a part of
+     *                              reconciliation of another resource.
      *
-     * @throws InvalidResourceException if the resource cannot be safely reconciled.
+     * @return  Set of conditions
      *
-     * @return set of conditions
+     * @throws InvalidResourceException if the resource cannot be safely reconciled.
      */
-    public static <T extends CustomResource<P, S>, P extends Spec, S extends Status> Set<Condition> validate(Reconciliation reconciliation, T resource) {
+    public static <T extends CustomResource<P, S>, P extends Spec, S extends Status> Set<Condition> validate(Reconciliation reconciliation, T resource, boolean printResourceName) {
```

Comment:
I was wondering if we really need an extra flag to decide to print (or not) the resource name.
When "not needed" it would be just redundant and obvious but does it really hurt? In all other case it would be useful (i.e. KafkaConnector, KafkaNodePool). Having this flag just makes the code more "complicated" imho.

Reply:
I can remove it if you think. But keep in mind that all other cases are the 8 other custom resources basically.

Reply:
Should be done now.

Reply:
Thanks!